bot/src/main/java/com/community/tools/service/github/event/stats/GithubWorkflowRunEventHandler.java

```diff
@@ -0,0 +1,83 @@
+package com.community.tools.service.github.event.stats;
+
+import com.community.tools.model.TaskStatus;
+import com.community.tools.model.stats.UserTask;
+import com.community.tools.model.stats.UserTaskId;
+import com.community.tools.repository.stats.UserTaskRepository;
+import com.community.tools.service.github.event.GithubEventHandler;
+import java.time.LocalDate;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class GithubWorkflowRunEventHandler implements GithubEventHandler {
+
+  private final UserTaskRepository userTaskRepository;
+
+  @Override
+  public void handleEvent(JSONObject eventJson) {
+    if (eventJson.has("action") && eventJson.getString("action").equals("completed")) {
+      parseAndSaveWorkflowRun(eventJson);
+    }
+  }
+
+  /**
+   * Processes JSON data from GitHub workflow_run event, saves it to the configured database.
+   *
+   * @param eventJson - representation of the JSON data from GitHub workflow_run event
+   * @see <a
+   * href="https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_run">GitHub docs
+   * about the event</a>
+   */
+  private void parseAndSaveWorkflowRun(JSONObject eventJson) {
+    final JSONObject repo = (JSONObject) eventJson.get("repository");
+    final JSONObject workflowRun = (JSONObject) eventJson.get("workflow_run");
+    final String gitName = ((JSONObject) workflowRun.get("actor")).getString(
+        "login");
+    final String taskName = getTaskName(repo.getString("name"), gitName);
+    final String repoFullName = repo.getString("full_name");
+    final String conclusion = workflowRun.getString("conclusion");
+    final int pullId = ((JSONObject) workflowRun.getJSONArray(
+        "pull_requests").get(0)).getInt("number");
```

Comment:
Will it brake if somebody creates their own PR's separately from autogenerated one?

Reply:
This could indeed have broken it, since it turns out this is just a list of open pull requests in the repo according to this [discussion](https://github.com/orgs/community/discussions/24674) I found, a pity it is not mentioned in the docs =(. But, I should not have made assumptions. Revised the logic quite a bit in the latest version.